src/parser/SGTypes.ts

```diff
@@ -323,61 +451,72 @@ export class SGInterface extends SGTag {
         field.alias = alias;
     }
 
+    removeField(id: string) {
+        for (let i = 0; i < this.childNodes.length; i++) {
+            const node = this.childNodes[i];
+            if (node.tagName === 'field' && node.id === id) {
+                this.childNodes.splice(i, 1);
+                return;
+            }
+        }
+    }
+
     getFunction(name: string) {
         return this.functions.find(field => field.name === name);
     }
+
+    /**
+     * Add or replace a function on the interface
+     */
     setFunction(name: string) {
         let func = this.getFunction(name);
         if (!func) {
-            func = new SGFunction();
-            func.name = name;
-            this.functions.push(func);
+            func = createSGInterfaceFunction(name);
+            this.childNodes.push(func);
         }
     }
 
-    protected transpileBody(state: TranspileState): (string | SourceNode)[] {
-        const body: (string | SourceNode)[] = ['>\n'];
-        state.blockDepth++;
-        if (this.fields.length > 0) {
-            body.push(...this.fields.map(node => node.transpile(state)));
-        }
-        if (this.functions.length > 0) {
-            body.push(...this.functions.map(node => node.transpile(state)));
+    /**
+     * Remove a function from the interface
+     */
+    removeFunction(name: string) {
+        for (let i = 0; i < this.childNodes.length; i++) {
+            const node = this.childNodes[i];
+            if (node.tagName === 'function' && node.getAttributeValue('name') === name) {
+                this.childNodes.splice(i, 1);
+                return;
+            }
         }
-        state.blockDepth--;
-        body.push(state.indentText, '</', this.tag.text, '>\n');
-        return body;
     }
 }
 
 export class SGComponent extends SGTag {
-    constructor(
-        tag: SGToken = { text: 'component' },
-        attributes?: SGAttribute[],
-        content?: SGTag[],
-        range?: Range
-    ) {
-        super(tag, attributes, range);
-        if (content) {
-            for (const tag of content) {
-                if (isSGInterface(tag)) {
-                    this.api = tag;
-                } else if (isSGScript(tag)) {
-                    this.scripts.push(tag);
-                } else if (isSGChildren(tag)) {
-                    this.children = tag;
-                } else {
-                    throw new Error(`Unexpected tag ${tag.tag.text}`);
-                }
-            }
+
+    /**
+     * Get all the <Field> and <Function> elements across all <Interface> nodes in this component
+     */
+    public get interfaceMembers() {
```

Comment:
API to find an `interface` node seem inconvenient (I assume it requires to use `getChildNodesByTagName<SGInterface>('interface')`). Adding fields/functions is going to be a major use case for the XML AST manipulations.

Reply:
The SGInterface element has these functions: `setField`, `removeField`, `setFunction`, `removeFunction`. However, you are right that authors would need to call `getChildNodesByTagName<SGInterface>('interface')` and use the first interface that they find. So what do you propose?  `setInterfaceField`, `removeInterfaceField`, `setInterfaceFunction`, `removeInterfaceFunction` functions on `SGComponent` that auto-picks the first `SGInterface` it finds, or creates one if missing?

Reply:
@elsassph  I went ahead and added a bunch of interface-related methods to SGComponent. [Here's the commit](https://github.com/rokucommunity/brighterscript/commit/52cc44ef3c06bc13a9c10c633a28f38f545dd5e1). Do these suit your needs?